perf/src/sigverify.rs

```diff
@@ -608,15 +610,38 @@ pub fn shrink_batches(batches: &mut Vec<PacketBatch>) {
 
 pub fn ed25519_verify_cpu(batches: &mut [PacketBatch], reject_non_vote: bool, packet_count: usize) {
     debug!("CPU ECDSA for {}", packet_count);
-    PAR_THREAD_POOL.install(|| {
-        batches.into_par_iter().for_each(|batch| {
-            batch.par_iter_mut().for_each(|packet| {
-                if !packet.meta.discard() && !verify_packet(packet, reject_non_vote) {
-                    packet.meta.set_discard(true);
-                }
-            })
+
+    let thread_count = packet_count
+        .saturating_add(VERIFY_MIN_PACKETS_PER_THREAD)
+        .saturating_div(VERIFY_MIN_PACKETS_PER_THREAD);
+    if thread_count >= get_thread_count() {
+        // When using all available threads, skip the overhead of flatting, collecting, etc.
+        PAR_THREAD_POOL.install(|| {
+            batches.into_par_iter().for_each(|batch| {
+                batch.par_iter_mut().for_each(|packet| {
+                    if !packet.meta.discard() && !verify_packet(packet, reject_non_vote) {
+                        packet.meta.set_discard(true);
+                    }
+                })
+            });
         });
-    });
+    } else {
+        let packets_per_thread = packet_count.saturating_div(thread_count);
+        PAR_THREAD_POOL.install(|| {
+            batches
+                .into_par_iter()
+                .flatten()
```

Comment:
Just for my understanding:

1. The goal of the flatten is to avoid the call to the `batches.into_par_iter()` which would wake up `batches.len()` number of threads? This is necessary when the batches are small.

2. If thread_count is less than the number of threads in the pool, does the `with_min_len` actually prevent waking up every thread in the pool? Seems based on your posted graph it does, disregarding the power of 2 behavior.